overlord/devicestate/handlers.go

```diff
@@ -99,9 +99,13 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	if m.preseed {
 		return fmt.Errorf("internal error: mark-seeded task not expected in pre-seeding mode")
 	}
-	// TODO:UC20: update "modeenv" and remove "recovery_system" from
-	// it because this information is only needed for the initial
-	// seeding.
+
+	// unset recovery_system because that is only needed during install mode
+	m.modeEnv.RecoverySystem = ""
+	err := m.modeEnv.Write("")
```

Comment:
we can't do this unconditionally, we don't want this file to show up on UC16/18. Also I think we want to keep some trail in state about good recovery systems we managed to seed but that part is not a blocker for this.

Reply:
ah yes you're right... sorry I forgot there's these things not known as uc20 we happen to run snapd on sometimes

Reply:
maybe grab a DeviceCtx (see handlers_install.go) and check HasModeenv and OperatingMode is "run" ?